### PR TITLE
catppuccin-whiskers: init at 2.4.0

### PR DESCRIPTION
**Pull request by NamesCode**
## Description of changes

Setup a `package.nix` for [whiskers](https://github.com/catppuccin/toolbox/tree/main/whiskers) and register myself as its maintainer.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by isabelroses** [comment]
Result of `nixpkgs-review pr 317021` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>catppuccin-whiskers</li>
  </ul>
</details>

**Review by isabelroses — APPROVED** [review]

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -14335,6 +14335,11 @@
     githubId = 1009523;
     name = "Ashijit Pramanik";
   };
+  Name = {
```
Comment (isabelroses) [inline maintainers/maintainer-list.nix:14338]:
```suggestion
  Name = {
    name = "Name";
```

**Review by uncenter — CHANGES_REQUESTED** [review]
Commit message and title should be `catppuccin-whiskers:` and not `whiskers:` too.

File: pkgs/by-name/ca/catppuccin-whiskers/package.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{ lib, fetchFromGitHub, rustPlatform, }:
```
Comment (jtbx) [inline pkgs/by-name/ca/catppuccin-whiskers/package.nix:1]:
```suggestion
{ lib, fetchFromGitHub, rustPlatform }:
```

File: pkgs/by-name/ca/catppuccin-whiskers/package.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    hash = "sha256-gxM2SZEc6RfV5XzSqCMxKqt7dPnsIIybl4B5bD66j6E=";
+  };
+
+  buildAndTestSubdir = "whiskers";
+  cargoHash = "sha256-sW0847nGUJVfJ9zzSb96h3jjCc0m3A/MIO0fzYrnA+A=";
+
+  meta = {
+    homepage = "https://github.com/catppuccin/toolbox/tree/main/whiskers";
+    description =
+      "A templating tool to simplify the creation of Catppuccin ports";
```
Comment (jtbx) [inline pkgs/by-name/ca/catppuccin-whiskers/package.nix:20]: This should all be on the same line

**Review by jtbx — APPROVED** [review]
Welcome!

File: pkgs/by-name/ca/catppuccin-whiskers/package.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "catppuccin";
+    repo = "whiskers";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-gxM2SZEc6RfV5XzSqCMxKqt7dPnsIIybl4B5bD66j6E=";
+  };
+
+  buildAndTestSubdir = "whiskers";
```
Comment (uncenter) [inline pkgs/by-name/ca/catppuccin-whiskers/package.nix:14]: This line shouldn't be needed anymore.

File: pkgs/by-name/ca/catppuccin-whiskers/package.nix [hunk 5]
```diff
@@ -0,0 +1,24 @@
+{ lib, fetchFromGitHub, rustPlatform }:
+let version = "2.3.0";
```
Comment (uncenter) [inline pkgs/by-name/ca/catppuccin-whiskers/package.nix:2]: `2.4.0` is now out.

File: pkgs/by-name/ca/catppuccin-whiskers/package.nix (hunk 4 above)
Comment (NamesCode) [reply pkgs/by-name/ca/catppuccin-whiskers/package.nix:14]: Ah my bad. Noticed this earlier but forgot to remove

**Review by isabelroses — APPROVED** [review]

**Review by uncenter — APPROVED** [review]

**Pull request merged**